### ISSUE TITLE
Python Layer

### ISSUE THREAD

**Issue by stonier** [maliput/delphyne#180]
This is a meta-issue identifying the tasks required for Milestone #2. Some more detailed notetaking [here](https://docs.google.com/document/d/1d_QeCL71XdVVZqWZmBc1ZFM81HCVtE0x2hQLG8K5mJ0/edit#).

_Scenarios as a python script which is responsible for instantiating and configuring the simulator, agents, callback and evaluation functions, as well as ultimately starting and stepping through the simulation._

**Simulator**

Be able to instantiate the simulator with options so that the same script can be used in both interactive and CI modes.

- [x] Instantiate the simulator
- [x] able to start the simulation in paused mode
- [x] able to start the simulation in 'as fast as possible' mode (i.e. non-realtime)
- [x] able to run the simulation for a finite duration
- [x] command-line options framework for pause, non-realtime, duration modes

**Road Network**

- [x] instantiate a dragway (proof of life) [Issue #270]
- [x] instantiate a multilane roadway (where we want to be)

**Agents**

- [x] add a simple car
- [x] add a lane following car
- [x] add a lane changing & merging (MOBIL) car

**Dynamic Scripting**

- [x] execution time statistic collection, dump to stdout at the end of the simulation [#333, #373]
- [x] ~vehicle collision notification and abort out of the simulation loop, exit with failure~ (moved to #372)

**Infra**

- [ ] Simulation Runner should not be compiled as `STATIC` [#341]
- [x] ~Switch to python3, #158~ (moved to #372)

**Documentation**

* [ ] User and Developer notes in the [Delphyne Guide](https://docs.google.com/document/d/1tQ9vDp084pMuHjYmtScLB3F1tdr4iP9w7_OTcoSM1zQ/edit#heading=h.875k18u2f3q0). Document the caveats in the developer's section (i.e. new type, expose it in `python_bindings.cc`)

**Demo Goal**

- [ ] Multilane world with lane changing cars and finite amount of time
- [x] ~One parked car (external), one simple car that collides into the parked car, aborts simulation~ (moved to #372)

-----

**Priorities**

From least to highest

* Vehicle collisions : might be some work if we go the c++ route, also pending what robotics does
* External cars : since it's blocking on Chris' work anyway
* Remaining : should all be unblocked and not needing any design

**Risks**

* The python bindings bucket since it's very beta and we may need many parameter bindings
* Not yet finalised how we transport maliput and view it in the visualisers
